python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
Ok. However, `_index_cache_dict`, `_registry` and  `_data_dict` are `GangaObject` properties, not `Node` properties so they should be moved to the `__slots__` attribute on that class.